Docs/Resource Manager.md

```diff
@@ -0,0 +1,308 @@
+The interface for this is hsResMgr::AddViaNotify().
+
+In all cases, the Resource Manager is at least an intermediary to the delivery of a plRefMsg.
+plRefMsg’s should not be sent directly through the dispatch system from one object to another.
+
+The defined contexts are as follows:
+
+* ***kOnCreate*** – The asset has just been created or loaded.
+  The receiver is free to cache and use its pointer until notified otherwise.
+  This context generally means the object has either just been created on scene conversion,
+  or has just been loaded into memory.
+  Typically from hsResMgr::ReadKeyNotifyMe().
+* ***kOnDestroy*** – The asset is about to be destroyed.
+  The receiver should discard any references to its pointer value that it may have cached.
+  This context usually indicates that the object is being paged out.
+  Typically sent automatically as notification when an object is destroyed.
+* ***kOnRequest*** – This is a request to add the asset reference, usually to a list of like objects.
+  Examples are adding a light to a SceneNode or a Modifier to a SceneObject.
+  This context usually implies that the plRefMsg did not originate from the receiver.
+  The plRefMsg might have come from the asset itself, or from a 3rd party manager object.
+  Typically from hsResMgr::AddViaNotify().
+* ***kOnRemove*** – The asset is requesting that it be removed, usually from a list of like objects.
+  This is typically the reversal of a previous kOnRequest message.
+  Typically from one plKey releasing another.
+* ***kOnReplace*** – An asset is being delivered as a replacement to another asset already received.
+  References to the specified old asset should be discarded and replaced by references to the new one.
+  Not currently used.
+
+### Current Limitations
+
+#### Keys for Dynamic Objects
+
+The current system uses UOID’s as the underlying identification method for plKey’s.
+The last field of the UOID is a sequence number,
+which is assigned at scene conversion time.
+This leaves no provision for objects created at run-time (procedural objects).
+There is a mechanism supplied (plTempKey),
+but these aren’t well integrated into the rest of the resource manager.
+The requirements for procedural objects are the abilities at run-time to:
+
+* Register with the Resource Manager.
+* Receive from the Resource Manager a plKey that other objects will know how to request.
+* Interact with non-procedural assets normally.
+* UnRegister from the Resource Manager at any time.
+* If still registered, delete at shutdown via the normal dependency mechanism.
+
+#### Identifiers Across Source Asset Files
+
+Again because the UOID’s are generated at scene conversion time,
+there is currently no way for an object in one source Max file to reference an object in another Max file.
+The requirement is that the Age/Chapter/Page/ClassID/&lt;string identifier> of an object may be efficiently converted at run-time into a plKey for the object.
```

Comment:
```suggestion
The requirement is that the Age/Chapter/Page/ClassID/<string identifier> of an object may be efficiently converted at run-time into a plKey for the object.
```

Reply:
Escaping the `<` as `&lt;` is necessary because Markdown supports inline HTML. Perhaps not all Markdown parsers implement this consistently, but at least GitHub and IntelliJ's parsers recognize `<string identifier>` as an HTML tag, so it becomes invisible if the `<` isn't escaped like this.

Reply:
In that case, let's escape both brackets to make it clearer that it's not an error 😄